docs/how-to/hip_runtime_api/call_stack.rst

```diff
@@ -0,0 +1,127 @@
+.. meta::
+    :description: This page describes call stack concept in HIP
+    :keywords: AMD, ROCm, HIP, call stack
+
+*******************************************************************************
+Call stack
+*******************************************************************************
+
+The call stack is a concept in computer science, representing the stack data
+structure that used for managing function calls, by saving the state of the
+current function. Each time a function is called, a new call frame is added to
+the top of the stack, containing information such as local variables, return
+addresses and function parameters. When the function execution completes, the
+frame is removed from the stack and loaded back into the corresponding
+registers. This concept allows the program to return to the calling function
+and continue execution from where it left off.
+
+The call stack for each thread must track its function calls, local variables,
+and return addresses. However, in GPU programming, the memory required to store
+the call stack increases due to the parallelism inherent to the GPUs. NVIDIA
+and AMD GPUs are using different approaches. NVIDIA GPUs have the independent
+thread scheduling feature since Volta GPU, where every thread has its own call
+stack and effective program counter. On AMD GPUs threads are grouped;
+therefore, every warp/wavefront has its own call stack and program counter, but
```

Comment:
```suggestion
each warp/wavefront has its own call stack and program counter, but
```

Reply:
Also, in general do we need both warp and wavefront here? Don't we use the term wavefront?